src/lens.jl

```diff
@@ -131,41 +140,69 @@ end
     )
 end
 
-struct ComposedLens{L1, L2} <: Lens
-    lens1::L1
-    lens2::L2
+struct ComposedLens{LO, LI} <: Lens
+    inner::LI
+    outer::LO
 end
 
 compose() = IdentityLens()
 compose(l::Lens) = l
 compose(::IdentityLens, ::IdentityLens) = IdentityLens()
 compose(::IdentityLens, l::Lens) = l
 compose(l::Lens, ::IdentityLens) = l
-compose(l1::Lens, l2 ::Lens) = ComposedLens(l1, l2)
-function compose(ls::Lens...)
+compose(inner::Lens, outer::Lens) = ComposedLens(inner, outer)
+function compose(l1::Lens, ls::Lens...)
     # We can build _.a.b.c as (_.a.b).c or _.a.(b.c)
     # The compiler prefers (_.a.b).c
-    compose(compose(Base.front(ls)...), last(ls))
+    compose(l1, compose(ls...))
 end
 
-function get(l::ComposedLens, obj)
-    inner_obj = get(l.lens2, obj)
-    get(l.lens1, inner_obj)
+"""
+    lens₁ ∘ lens₂
+    compose([lens₁, [lens₂, [lens₃, ...]]])
+
+Compose lenses `lens₁`, `lens₂`, ..., `lensₙ` to access nested objects.
+
+# Example
+```jldoctest
+julia> using Setfield
+
+julia> obj = (a = (b = (c = 1,),),);
+
+julia> la = @lens _.a
+       lb = @lens _.b
+       lc = @lens _.c
+       lens = la ∘ lb ∘ lc
+(@lens _.a.b.c)
+
+julia> get(obj, lens)
+1
+```
+"""
+Base.:∘(l1::Lens, l2::Lens) = compose(l1, l2)
+
+function get(obj, l::ComposedLens)
+    inner_obj = get(obj, l.inner)
+    get(inner_obj, l.outer)
```

Comment:
In composed lens, you named the field `outer`, that should be named `inner` and vice versa. For instance this code is much less confusing, if written like this:
```julia
function get(obj, l::ComposedLens)
    outer_obj = obj  # for clarity
    inner_obj = get(outer_obj, l.outer)
    get(inner_obj, l.inner)
end
```

Reply:
Hmm so my interpretation was `inner::Lens` means "lens to get the inner object" but I suppose what you meant was "lens applied to the inner object", right?  But to be honest thought about the second interpretation too.  I started wondering maybe we should call them `.left` and `.right`.

Reply:
Ah, never mind, I was somehow thinking left is inner. Left is obviously outer.

Reply:
Yes the inner/outer lens acts on the inner/outer object. Another reason is that if you have a function composition `g∘f` then `g` is called outer and `f` is called inner function (probably motivated by the formula `(g∘f)(x) = g(f(x))` where `g` resp. `f` appear outside resp. inside the parenthesis). So why not use the same terminology for composition in other categories.

Reply:
We also need the notation `f; g` for those who as confused as I was :)